catalog/rocks/cataloger.go

```diff
@@ -552,6 +552,12 @@ func (c *cataloger) Merge(ctx context.Context, repository string, leftBranch str
 	rightBranchID := graveler.BranchID(rightBranch)
 	meta := graveler.Metadata(metadata)
 	commitID, err := c.EntryCatalog.Merge(ctx, repositoryID, leftRef, rightBranchID, committer, message, meta)
+	if errors.Is(err, graveler.ErrConflictFound) {
+		// for compatibility with old cataloger
+		return &catalog.MergeResult{
+			Summary: map[catalog.DifferenceType]int{catalog.DifferenceTypeConflict: 1},
```

Comment:
Shouldn't `MergeResult` hold a _summary_ of the merge?  Shouldn't this be the count of what happened during the merge?

Reply:
Yes, we didn't implement this yet. So in case of a successful merge, the summary would be empty. It's an issue, but no a release-blocker (IMO). Will open an issue for that
In case of conflict, the MVCC would return a summary of {Conflict:1}, regardless of the number of conflicts. This PR aligns with this behavior.

Reply:
Cool, thanks!